synfig-studio/src/gui/modules/mod_palette/dock_paledit.cpp

```diff
@@ -349,6 +369,70 @@ Dock_PalEdit::add_color(const synfig::Color& x)
 	return size()-1;
 }
 
+bool
+Dock_PalEdit::check_hex_format(const gchar* hexcolor)
+{
+	//Check if clipboard is empty
+	if(hexcolor==NULL)
+		return false;
+
+	//Check if first character is a # for the correct format
+	if(hexcolor[0]!='#')
+		return false;
+
+	int strsize = strlen(hexcolor);
+
+	//Check size is correct for the format: #ffffff or #fff or #f
+	//these are the sizes supported by the Color::set_hex() function 
+	if(strsize!=7&&strsize!=4&&strsize!=2)
+		return false;
+
+	//Checking if all characters are between 0-F uppercase or lowercase in ascii
+	for(int i=1;i<strsize;i++)
+		if(!std::isxdigit(hexcolor[i]))
+			return false;
+
+	return true;
+}
+
+void
+Dock_PalEdit::add_from_clipboard()
+{
+	if(clipboard_==NULL)
+		clipboard_ = gtk_clipboard_get(GDK_SELECTION_CLIPBOARD);
+	
+	gchar* hexcolor = gtk_clipboard_wait_for_text(clipboard_);
+
+	//Clipboard is in hexformat. Add the color.
+	if(check_hex_format(hexcolor)){
+		std::string strcol = hexcolor;
+		strcol = strcol.substr(1,6);
+
+		synfig::Color col;
+		col.set_hex(strcol);
+		col.set_a(1.0f);
+
+		palette_.push_back(col);
+	}
+
+	//Free hexcolor string
+	g_free(hexcolor);
+
+	signal_changed()();
+	refresh();
+}
+
+void
+Dock_PalEdit::copy_color(int i)
+{
+	//Taking first 7 characters of the color string as they contain the formated #hexadecimal color
+	GtkClipboard* clipboard = gtk_clipboard_get(GDK_SELECTION_CLIPBOARD);
```

Comment:
Why do you use Gtk instead of gtkmm? :)
`Glib::RefPtr<Gtk::Clipboard> refClipboard = Gtk::Clipboard::get();`
https://developer-old.gnome.org/gtkmm-tutorial/3.24/sec-clipboard-copy.html.en
https://developer-old.gnome.org/gtkmm/3.24/classGtk_1_1Clipboard.html

Reply:
oh thanks for the heads up! I wasn't aware that I should be using specifically gtkmm. I have changed the implementation to use the gtkmm functions from those links :).